github/spec/features/validate_repos_spec.rb

```diff
@@ -0,0 +1,88 @@
+
+  end
+
+  it "should alert if finds an untagged repo in repos.json" do
+    
+    repos = [
+      { "app_name": "this-is-a-govuk-repo", },
+      { "app_name": "this-govuk-repo-is-not-tagged!", }
+  ]
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-govuk-repo")
+    octokit_mock = double("Octokit::Client", org_repos: [repo_mock])
+
+    stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
+      to_return(status: 200, body: repos.to_json, headers: {})
+
+    expect { ValidateRepos.new(octokit_mock).verify_repo_tags }.to output("\e[0;31;40mUntagged govuk repos:\e[0m\nthis-govuk-repo-is-not-tagged!\n\n\e[0;32;40mFalsely tagged govuk repos: No mismatches found.\e[0m\n").to_stdout
+
+  end
+
+  it "should alert if it finds a repo that has falsely been tagged as govuk." do
+    
+    repos = []
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-falsley-tagged-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-falsley-tagged-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-falsley-tagged-govuk-repo")
+    octokit_mock = double("Octokit::Client", org_repos: [repo_mock])
+
+    stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
+      to_return(status: 200, body: repos.to_json, headers: {})
+
+    expect { ValidateRepos.new(octokit_mock).verify_repo_tags }.to output("\e[0;32;40mUntagged govuk repos: No mismatches found.\e[0m\n\n\e[0;31;40mFalsely tagged govuk repos:\e[0m\nthis-is-a-falsley-tagged-govuk-repo\n").to_stdout
+
+  end
+
+  it "should alert reports findings accordingly and appropriately when it finds both." do
+    
+    repos = [{
+       "app_name": "this-govuk-repo-is-not-tagged!", 
+       }]
+
+    repo_mock = double("Repo", topics: ["govuk"], archived: false)
+    allow(repo_mock).to receive(:[]).with("name").and_return("this-is-a-falsley-tagged-govuk-repo")
+    allow(repo_mock).to receive(:full_name).and_return("this-is-a-falsley-tagged-govuk-repo")
+    allow(repo_mock).to receive(:[]).with(:full_name).and_return("this-is-a-falsley-tagged-govuk-repo")
+    octokit_mock = double("Octokit::Client", org_repos: [repo_mock])
+
+    stub_request(:get, "https://docs.publishing.service.gov.uk/repos.json").
+      to_return(status: 200, body: repos.to_json, headers: {})
+
+    expect { ValidateRepos.new(octokit_mock).verify_repo_tags }.to output("\e[0;31;40mUntagged govuk repos:\e[0m\nthis-govuk-repo-is-not-tagged!\n\n\e[0;31;40mFalsely tagged govuk repos:\e[0m\nthis-is-a-falsley-tagged-govuk-repo\n").to_stdout
```

Comment:
This test will also still work with the refactor described above, but again you'd just need to change `this-is-a-falsley-tagged-govuk-repo` to `this-is-a-govuk-repo`.